pkg/catalog/retry.go

```diff
@@ -0,0 +1,74 @@
+package catalog
+
+import (
+	"github.com/openservicemesh/osm/pkg/apis/policy/v1alpha1"
+	"github.com/openservicemesh/osm/pkg/identity"
+	"github.com/openservicemesh/osm/pkg/policy"
+	"github.com/openservicemesh/osm/pkg/service"
+)
+
+// GetRetryPolicy returns the RetryPolicySpec for the given downstream identity and upstream service
+// If a source and destination has more than one retry policy that can be applied, the more
+// specific retry policy will be used. Types of retry policy:
+// - specific retry policy that specifies the source and destination
+// - wildcard destination service retry policy
+// - wildcard destination service and namespace retry policy
+func (mc *MeshCatalog) GetRetryPolicy(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService) *v1alpha1.RetryPolicySpec {
```

Comment:
The function should not be a part of the `MeshCataloger` interface and instead be invoked while building the routes within the existing catalog APIs.
```suggestion
func (mc *MeshCatalog) getRetryPolicy(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService) *v1alpha1.RetryPolicySpec {
```

Reply:
+1